mood-analysis.rb

```diff
@@ -19,7 +19,57 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def strip_punctuation(string)
+  string.gsub!(/[^0-9A-Za-z]/, ' ')
+end
+
+def happy_days(list_of_moods)
+  # how many tweets until there have been 3 happy days?
+  number_of_happy_days = 0
+  moods_read = 0
+  list_of_moods.each do |mood|
+    # THANK YOU RUBY KOANS!!!
+    break moods_read if number_of_happy_days == 3
+    number_of_happy_days += 1 if mood == ":-)"
+    moods_read += 1
+  end
+  if number_of_happy_days >= 3
+    return "It takes #{moods_read} entries for 3 happy days to occur"
+  else
+    return "Only #{number_of_happy_days} happy day(s)...bummer!"
+  end
+end
+
+def overall_mood(list_of_moods)
+  # determines most common mood across all tweets
+  # "The most common mood is :-)"
+
+  mood_hash = {
+    happy: [],
+    sad: [],
+    neutral: []
+  }
+
+  list_of_moods.each do |mood|
+    if mood == ":-)"
+      mood_hash[:happy] << mood
+    elsif mood == ":-("
+      mood_hash[:sad] << mood
+    elsif mood == ":-|"
+      mood_hash[:neutral] << mood
+    end
+  end
+
+  max_pair = mood_hash.max_by { |key, value|
+    value.length
+  }
+
+  "The most common mood is #{max_pair[1][0]}"
+end
```

Comment:
I think my instructions say to return the most common mood, which I would have expected as an emoticon. Also, again, we want to return the smallest unit of inofmration possible, so not a string, just an emoticon.